### DISCUSSION TITLE
自定义UI

### DISCUSSION THREAD

**Discussion by hantengc** [rime/home#1360] [Q&A]
Linux上rime输入法有自己的UI界面吗？我看目前是fcitx5本身的UI界面

**Comment by hantengc** [comment]
@lotem 辛苦帮忙解答一下~~~

**Reply by MokOopsing** [reply]
所有的界面都由前端提供，librime仅作为后端

**Reply by hantengc** [reply]
window上也是一样的逻辑嘛？我看windows上也有rime输入方案UI，并且和系统其他的输入法UI不一样

**Comment by lotem** [comment]
无有。

**Comment by hantengc** [comment]
rime的输入方案中
switcher：
  -name：ascii_mode
   reset：0
   states：[中文，西文]

这个在每次切换输入法的时候都会显示，我能不能设置不让他们显示呢？

**Reply by hantengc** [reply]
@lotem @mokapsing 辛苦帮忙解答一下 谢谢~

**Reply by MokOopsing** [reply]
除非你干掉这个开关

**Reply by hantengc** [reply]
干掉这个开关？那我是不是就没办法设置全角半角了？

**Reply by hantengc** [reply]
包括其他的在这个地方被设置的功能

**Reply by MokOopsing** [reply]
这个和全角半角没什么关系吧，另外你可以用shift切换ascii_mode，这里的开关和你用按键切换没什么关系

**Reply by MokOopsing** [reply]
![image](https://github.com/rime/home/assets/39441028/0bfbeb2e-808b-4755-b35b-c3c7cce813f4)

**Reply by hantengc** [reply]
在哪里定义了shift切换ascii_mode呢？我想实现的是快捷键切换中西文、全角半角、简体繁体等。小白不太懂在哪里设置为快捷键切换

**Reply by hantengc** [reply]
因为我经常使用中文输入法例如搜狗，在UI界面上就可以选择全角半角。或者通过快捷键切换简体繁体。
我现在觉着候选2有点多余想去掉，但是功能还希望继续保存。通过快捷键切换也是一种方式

**Reply by MokOopsing** [reply]
小白就去看看说明书，你的问题太多了，我回答不过来

https://github.com/LEOYoon-Tsaw/Rime_collections/blob/master/Rime_description.md

**Reply by hantengc** [reply]
thanks.
